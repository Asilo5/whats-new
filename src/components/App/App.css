* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.app {
  display: flex;
  background: #B4A3B7;
  text-align: center;
  width: 100%;
}

.main-news {
  width: 100%;
}

