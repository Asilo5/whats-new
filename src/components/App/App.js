import React, { Component } from 'react';
import local from '../../data/local';
import './App.css';
import Menu from '../Menu/Menu';
import SearchForm from '../SearchForm/SearchForm';
import NewsContainer from '../NewsContainer/NewsContainer';


class App extends Component {
  constructor() {
    super();
    this.state = {
      local: ''
    }
  }

  render () {
    return (
      <section className="app">
          <Menu />
          <section className='main-news'>
            <SearchForm />
            <NewsContainer />
          </section>
      </section>
    );
  }
}

export default App;
