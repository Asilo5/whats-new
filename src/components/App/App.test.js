import React from 'react';
import { shallow } from 'enzyme';
import App from './App';


describe('App',() => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  })

  it('should update state when changeNews is called', () => {
    const mockArticle = {
      description: 'The web slinger',
      headline: 'Spider-Man ',
      id: 1,
      img: 'https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp',
      url: 'https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html'
    }

    expect(wrapper.state('news')).toEqual('');
    wrapper.instance().changeNews(mockArticle);

    expect(wrapper.state('news')).toEqual(mockArticle);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

})

