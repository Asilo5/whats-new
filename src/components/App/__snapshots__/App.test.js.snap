// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<section
  className="app"
>
  <Menu
    allNews=""
    changeNews={[Function]}
  />
  <section
    className="main-news"
  >
    <SearchForm
      changeNews={[Function]}
      findNews={[Function]}
    />
  </section>
</section>
`;
